service/src/main/java/tds/assessment/repositories/impl/AssessmentQueryRepositoryImpl.java

```diff
@@ -245,52 +248,30 @@ public AssessmentQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTempla
     }
 
     @Override
-    public Optional<Assessment> findAssessmentBySegmentKey(final String segmentKey) {
-        SqlParameterSource parameters = new MapSqlParameterSource("segmentKey", segmentKey);
+    public Optional<SegmentMetadata> findSegmentMetadata(final String segmentKey) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("segmentKey", segmentKey);
 
-        String SQL =
-            ASSESSMENT_SELECT_SQL +
-                "FROM \n" +
-                "   itembank.tblsetofadminsubjects A \n" +
-                "LEFT JOIN tblsetofadminsubjects B \n" +
-                "   ON (A.virtualtest = B.virtualtest OR B.virtualtest = A._key) \n" +
-                "JOIN \n" +
-                "   configs.client_testproperties CT \n" +
-                "   ON CT.testid = A.testid \n " +
-                "   OR CT.testid = (\n" +
-                "       SELECT parentTsa.testid \n" +
-                "       FROM itembank.tblsetofadminsubjects tsa \n" +
-                "       JOIN itembank.tblsetofadminsubjects parentTsa ON tsa.virtualtest = parentTsa._key \n" +
-                "       WHERE tsa._key = A._key \n" +
-                "   ) \n" +
-                "LEFT JOIN \n" +
-                "   configs.client_segmentproperties SP \n" +
-                "   ON SP.segmentid = A.testid \n" +
-                "LEFT JOIN \n" +
-                "   itembank.tblsubject S \n" +
-                "   ON S._key = A._fk_Subject \n" +
-                "JOIN \n" +
-                "   itembank.tblclient CL \n" +
-                "   ON (CL.name = SP.clientname OR CL.name = CT.clientname) \n" +
-                "LEFT JOIN \n" +
-                "   itembank.tbltestadmin TA \n" +
-                "   ON TA._fk_client = CL._key \n" +
-                "WHERE \n" +
-                "   B._key = :segmentKey \n OR A._key = :segmentKey \n" +
-                "ORDER BY  \n" +
-                "   assessmentKey DESC, \n" +
-                "   A.testPosition";
-
-        List<Map<String, Object>> rows = jdbcTemplate.queryForList(SQL, parameters);
-        Optional<Assessment> maybeAssessment = Optional.empty();
+        final String SQL = "SELECT \n" +
```

Comment:
Makes sense to me ... i think 2 easy queries might even be faster in practice than 1 complex one